iothub/device/samples/DeviceReconnectionSample/ExponentialBackoffTransientExceptionRetryPolicy.cs

```diff
@@ -30,9 +30,9 @@ public bool ShouldRetry(int currentRetryCount, Exception lastException, out Time
         {
             if (currentRetryCount < _maxRetryCount)
             {
-                if ((lastException is IotHubException iotHubException && iotHubException.IsTransient)
+                if ((lastException is IotHubClientException iotHubException && iotHubException.IsTransient)
                     || ExceptionHelper.IsNetworkExceptionChain(lastException)
-                    || (_exceptionsToBeIgnored != null && _exceptionsToBeIgnored.ContainsKey(lastException.GetType())))
+                    || (_exceptionsToBeIgnored != null && _exceptionsToBeIgnored.ContainsKey(((IotHubClientException)lastException).StatusCode)))
```

Comment:
May need to rearrange the order of these logical to access the confirmation that `lastException is IotHubClientException iotHubException`.
```suggestion
                    || (_exceptionsToBeIgnored != null && _exceptionsToBeIgnored.ContainsKey(iotHubException.StatusCode))
```